frontend/src/app/shared/components/calendar/calendar.component.ts

```diff
@@ -1,14 +1,57 @@
-import { Component, Input } from '@angular/core';
+import { Component, forwardRef, Input } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
 
 @Component({
     selector: 'app-calendar',
     templateUrl: './calendar.component.html',
     styleUrls: ['./calendar.component.sass'],
+    providers: [
+        {
+            provide: NG_VALUE_ACCESSOR,
+            // eslint-disable-next-line no-use-before-define
+            useExisting: forwardRef(() => CalendarComponent),
+            multi: true,
+        },
+    ],
 })
-export class CalendarComponent {
+
+export class CalendarComponent implements ControlValueAccessor {
```

Comment:
why do we need ControlValueAccessor?

Reply:
To set and get date value in calendar

Reply:
I've tried to find the way for setting and getting the value of calebdar and found thi artice https://medium.com/@majdasab/implementing-control-value-accessor-in-angular-1b89f2f84ebf. Actually, I'm not sure fi it's the best aproach but it work. What do you think about it? Pls suggest.

Reply:
Find a few ways to implement this and I will advice which one is the best for us

Reply:
will look into it

Reply:
I think calendar was out of the scope of my task. I wanted to do it, but it seems I don't have enough skills to implement it. Could you pls assign this part on someone who has more skills. @Anatolii-Ivanov

Reply:
I've found this article https://stackoverflow.com/questions/59049782/angular-controlvalueaccessor-vs-input-when-to-use-what-with-forms and it proves that my aproach is prefereble.